app/src/main/java/com/sun/noteapp/ui/search/SearchPresenter.kt

```diff
@@ -0,0 +1,36 @@
+package com.sun.noteapp.ui.search
+
+import com.sun.noteapp.data.model.Note
+import com.sun.noteapp.data.repository.NoteLocalRepository
+import com.sun.noteapp.data.source.OnDataModifiedCallback
+import java.lang.Exception
+
+class SearchPresenter(
+    private val view: SearchContract.View,
+    private val repository: NoteLocalRepository
+) : SearchContract.presenter {
+
+    private val notes = mutableListOf<Note>()
+
+    override fun getAllNotes() {
+        repository.getAllNotes(object : OnDataModifiedCallback<List<Note>> {
+            override fun onSuccess(data: List<Note>) {
+                notes.addAll(data)
+            }
+
+            override fun onFailed(exception: Exception) {
+                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+            }
+        })
+    }
+
+    override fun searchNote(word: String) {
+        val searchNotes = mutableListOf<Note>()
+        if (word.isNotEmpty())
+            for (note in notes) {
```

Comment:
Đoạn này anh có thể dùng map collection 
```suggestion
            searchNotes = notes.map{}
```

Reply:
đoạn này dùng map là sao nhỉ a chưa hiểu ý lắm